poweractions/poweractions.py

```diff
@@ -0,0 +1,231 @@
+                colour=await ctx.embed_colour(),
+            )
+            embed.set_footer(text="Page {num}/{total}".format(num=idx, total=len(pages)))
+            embed_pages.append(embed)
+        await menus.menu(ctx, embed_pages, menus.DEFAULT_CONTROLS)
+
+    @checks.admin()
+    @commands.command()
+    async def restartserver(self, ctx: commands.Context, server) -> None:
+        """
+        Restarts a server.
+        """
+        async with ctx.typing():
+            selectedserver = await self.config.guild(ctx.guild).servers()
+
+            if server not in selectedserver:
+                await ctx.send("That server does not exist.")
+                return
+            else:
+                server = selectedserver[server]
+
+            authheader = "Basic " + base64.b64encode(f"{server['key']}:{server['token']}".encode("ASCII")).decode(
+                "ASCII")
+
+            try:
+                async with aiohttp.ClientSession() as session:
+                    async def load():
+                        async with session.post(server["address"] + f"/instances/{server['key']}/restart",
+                                                headers={"Authorization": authheader}) as resp:
+                            if resp.status != 200:
+                                await ctx.send(f"Failed to restart the server. Wrong status code: {resp.status}")
+                                return
+
+                    await asyncio.wait_for(load(), timeout=5)
+
+            except asyncio.TimeoutError:
+                await ctx.send("Server timed out.")
+                return
+
+            except Exception as e:
+                await ctx.send(f"Server failed to restart. Error: {e}")
+
+            await ctx.send("Server restarted successfully.")
+
+    @checks.admin()
+    @commands.command()
+    async def restartnetwork(self, ctx: commands.Context) -> None:
+        """
+        Attemps to restarts all servers on the bot.
+        """
+        view = ConfirmView(ctx.author, disable_buttons=True, timeout=30)
+        view.message = await ctx.send(":warning: You are about to restart all servers configured on this bot "
+                                      "instance, are you certain this is what you want to do", view=view)
+        await view.wait()
+        if view.result:
+            await ctx.send("Restarting all servers...")
+            async with ctx.typing():
+                network_data = await self.config.guild(ctx.guild).servers()
+
+                embed = Embed(title="Network Restart", description="Results of the restarts", color=await ctx.embed_colour())
+
+                for server_name, server_details  in network_data.items():
+                    authheader = "Basic " + base64.b64encode(f"{server_details['key']}:{server_details['token']}".encode("ASCII")).decode(
+                        "ASCII")
+
+                    try:
+                        async with aiohttp.ClientSession() as session:
```

Comment:
You should not create a new client session for every restart call, share it outside the for loop.